src/ClockPanel.tsx

```diff
@@ -258,71 +215,60 @@ class UnthemedClockPanel extends PureComponent<Props, State> {
     );
   }
 
-  renderDate() {
-    const { now } = this.state;
-    const { dateSettings } = this.props.options;
+  function renderDate() {
+    const { dateSettings } = props.options;
 
     const className = css`
       font-size: ${dateSettings.fontSize};
       font-weight: ${dateSettings.fontWeight};
     `;
 
     const display = now.locale(dateSettings.locale || '').format(dateSettings.dateFormat);
-    
+
     return (
       <span>
         <h3 className={className}>{display}</h3>
       </span>
     );
   }
 
-  renderTime() {
-    const { now } = this.state;
-    const { options } = this.props;
-    const { timeSettings, mode } = options;
+  function renderTime() {
```

Comment:
Wouldn't it make sense to extract these nested functions into their own components?

Reply:
Maybe, but this is a simple one file component. I didn't feel the need to overcomplicate it